src/Datadog.Trace.ClrProfiler.Native/rejit_handler.h

```diff
@@ -119,6 +133,47 @@ class RejitHandler {
   HRESULT NotifyReJITCompilationStarted(FunctionID functionId, ReJITID rejitId);
   void _addFunctionToSet(FunctionID functionId,
                          RejitHandlerModuleMethod* method);
+  
+  void EnqueueForRejit(size_t length, ModuleID* moduleIds, mdMethodDef* methodDefs) {
+    rejit_queue_->push(RejitItem((int)length, moduleIds, methodDefs));
+  }
+
+  void Shutdown() {
+    rejit_queue_->push(RejitItem(-1, nullptr, nullptr));
+    if (rejit_queue_thread_->joinable()) {
+      rejit_queue_thread_->join();
+    }
+  }
+
+ private:
+  static void enqueue_thread(RejitHandler* handler) {
+    auto queue = handler->rejit_queue_;
+    auto profilerInfo = handler->profilerInfo;
+
+    Info("Initializing ReJIT request thread.");
+    HRESULT hr = profilerInfo->InitializeCurrentThread();
+    if (FAILED(hr)) {
+      Warn("Call to InitializeCurrentThread fail.");
```

Comment:
Is there anything different we need to do when this fails? Probably the answer is no, but I thought I would ask

Reply:
Well I don't think we can do anything in that case.